service/topology_coordinator.cc

```diff
@@ -2224,7 +2224,7 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
     {}
 
     // Returns true if the upgrade was done, returns false if upgrade was interrupted.
-    future<bool> maybe_run_upgrade();
+    future<bool> maybe_run_upgrade(qos::service_level_controller& sl_controller);
```

Comment:
why migrate_to_v2 can't be made static? does  it depend on anything from service_level_controller class? this way there would be no need for this dependency

Reply:
right, it can be